new-system-tests/src/test/java/io/apicurio/registry/systemtests/SqlKeycloak.java

```diff
@@ -0,0 +1,108 @@
+package io.apicurio.registry.systemtests;
+
+import io.apicur.registry.v1.ApicurioRegistry;
+import io.fabric8.junit.jupiter.api.KubernetesTest;
+import io.fabric8.junit.jupiter.api.LoadKubernetesManifests;
+import io.fabric8.openshift.client.OpenShiftClient;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static io.apicurio.registry.systemtests.Utils.findRegistryOperatorDeployment;
+import static io.apicurio.registry.systemtests.Utils.isDeploymentReady;
+import static io.apicurio.registry.systemtests.Utils.waitDeploymentReady;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@KubernetesTest
+@LoadKubernetesManifests({
+        // Keycloak IAM
+        "/keycloak/00_service.yaml", // Service
+        "/keycloak/01_realm.yaml", // Keycloak realm as realm.json in ConfigMap
+        "/keycloak/02_deployment.yaml", // Deployment of Keycloak itself
+        // PostgreSQL database resources
+        "/sql/00_service.yaml", // Service
+        "/sql/01_deployment.yaml", // Deployment
+        // Apicurio Registry operator
+        "/apicurio/00_operator_group.yaml", // Operator group for Apicurio Registry operator
+        "/apicurio/01_subscription.yaml", // Apicurio Registry operator subscription
+        // Apicurio Registry instance
+        "/apicurio/02_registry_sql_keycloak.yaml" // Apicurio Registry instance with PostgreSQL storage and Keycloak IAM
+})
+public class SqlKeycloak extends TestBase {
+    /** {@link OpenShiftClient} instance for tests. */
+    private OpenShiftClient client;
+
+    /**
+     * @return {@link OpenShiftClient} instance used in tests.
+     */
+    @Override
+    protected OpenShiftClient getClient() {
+        return client;
+    }
+
+    /**
+     * Performs actions that should be done before all tests.
+     */
+    @BeforeAll
+    public static void beforeAll() {
+        // Log information about current action
+        System.out.println("Before all.");
+
+        // Client is not accessible here
+    }
+
+    /**
+     * Performs actions that should be done after all tests.
+     */
+    @AfterAll
+    public static void afterAll() {
+        // Log information about current action
+        System.out.println("After all.");
+
+        // Client is not accessible here
+    }
+
+    /**
+     * Performs actions that should be done before each test.
+     */
+    @BeforeEach
+    public void beforeEach() {
+        // Log information about current action
+        System.out.println("Before each.");
```

Comment:
Can we use DI and a logging library? So that ppl can decide how to configure it on their own.

Reply:
Yes, I am going to update it.

Reply:
Please check https://github.com/Apicurio/apicurio-registry/pull/2987/commits/fd889d4a4f76f05fd9c8d7ce86be89503444f215 if it looks at least a little bit as expected :) Thanks!

Reply:
yup, whatever fits, no strong preferences 👍